### ISSUE TITLE
Some apps are failing to install id `Sign apk before install` is enabled

### ISSUE THREAD

**Issue by soymadip** [MuntashirAkon/AppManager#1080]
### Please check before submitting an issue

- [X] I know what my device, OS and App Manager versions are
- [X] I know how to take logs
- [X] I know how to reproduce the issue which may not be specific to my device

### Describe the bug

Some apps are failing to install if 'Sign apk before install' is enabled, they are showing ```INSTALL_PARSE FAILED NOT APK: Failed to parse /[data/app/vmd/1905902452.tmp/](http://data/app/vmd/1905902452.tmp/) M Base.apk: Failed to load asset path /data/ [app/vmd/1905902452.tmp/Base.apk](http://app/vmd/1905902452.tmp/Base.apk)``` message .

but it installs if we use system package installer.....

### To Reproduce

- this is not happening with all apks, i faced with apks like https://cdn.discordapp.com/attachments/1091796292973375589/1111741445951471717/app-debug.apk (this is an official beta build of an popular open source anime app called [saikou](https://github.com/saikou-app/saikou)) or ExteraGram [latest build](https://telegram.me/exteraGramCI/228)
- Then - 
1. download it 
2. try to install it with App Manager
3. MUST enable 'sign app before installing' option.
4.  Now the installation failure popup will be shown.
5. now try to install it with android package manager, it will be successfully installed.

### Expected behavior

the apk will be installed successfully 

### Screenshots

![Screenshot_2023-05-27-20-52-12-95_718a221d6c53e9d44e23988b91117061](https://github.com/MuntashirAkon/AppManager/assets/84225810/b21525c0-829e-49a9-990a-0a68484b8bd9)

### Logs

[LogCat.txt](https://github.com/MuntashirAkon/AppManager/files/11585444/LogCaat.txt)

### Device info

- Device: Realme 9 5g SE [RMX3461]
- OS Version: Realme UI 4 (A13)
- App Manager Version: 3.2.0 - Alpha1
- Mode: NonRoot

### Additional context

I also tried with latest AppManager CI build from telegram and got the same result

**Comment by MuntashirAkon** [comment]
Thanks for reporting the issue. Logs in no-root mode didn't help much. But the steps to reproducible the issue did. I've found the following interesting lines in the logs:

```
ziparchive              system_server                        W  Zip: 2042 extraneous bytes at the end of the central directory
system_server           system_server                        E  Failed to open APK '/data/app/vmdl1501213821.tmp/Base.apk' Invalid file
```

These lines claim that Android is unable to read the signed APK file properly which is weird. Because when I attempted to install that file manually again, it worked fine. So, it seems it has something to do with accessing the file itself rather than signing or anything else. This requires a bit more investigation to figure out the underlying cause though.

**Comment by soymadip** [comment]
> Thanks for reporting the issue. Logs in no-root mode didn't help much. But the steps to reproducible the issue did. I've found the following interesting lines in the logs:
> 
> ```
> ziparchive              system_server                        W  Zip: 2042 extraneous bytes at the end of the central directory
> system_server           system_server                        E  Failed to open APK '/data/app/vmdl1501213821.tmp/Base.apk' Invalid file
> ```
> 
> These lines claim that Android is unable to read the signed APK file properly which is weird. Because when I attempted to install that file manually again, it worked fine. So, it seems it has something to do with accessing the file itself rather than signing or anything else. This requires a bit more investigation to figure out the underlying cause though.

This may not be the case but ti guess i should inform, the apks i faced problems are mostly CI builds of the apps..

**Comment by MuntashirAkon** [comment]
Fixed in 61a51f5ebc48fb43222ed348ea10122f4c17d6ca
